### DISCUSSION TITLE
Filter specific terraform module based on the source for raising PR

### DISCUSSION THREAD

**Discussion by rafat-parveen** [renovatebot/renovate#21630] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

35.48

### If you're self-hosting Renovate, select which platform you are using.

gitlab.com

### If you're self-hosting Renovate, tell us what version of the platform you run.

GitLab Enterprise Edition 15.11.0-pre

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

We want to run renovate for dependecies scanning based on the Event.

The Event would be when any new tag is generated for particular repo then all the place where the repo is referenced, the tag shoud be updated there.

Example:
let say one repository name: terraform-aws-s3-bucket I have updated some code on it and once my PR is merged, it generate new tag.
once any new tag is generated for this repository, I have to update it's new tag in all the repo where this terraform-aws-s3-bucket is used. ( suppose new tag now v2.0.0).

another repository called, aws-static-site-blueprint
has multiple terraform file, s3-bucket.tf, lamda.tf, xyz.tf

In s3-bucket.tf, we use module to reuse the code from other repos (terraform-aws-s3-bucket ) and called it like this,
module “s3” { source = "git@gitlab.com:repo/blueprint/terraform-aws-s3-bucket?ref=v1.0.0" }

lamda.tf
module “lamda” { source = "git@gitlab.com:repo/blueprint/terraform-aws-lamda?ref=v1.0.0" }

so now I want to update the tag on s3 to v2.0.0 as this s3 repo generated new tag by my PR merge, so I want to automate this update via renovate, but renovate update the tag in all .tf file where it find update.

so I want to filter based on the source of terrfaorm module where dependencies shoud be updated based on which repo is merged and generated new tag at the moment.
### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by rafat-parveen** [comment]


**Reply by rarkins** [reply]
No

**Reply by rafat-parveen** [reply]
This means Renovate doesn't support above feature, Could you please clarify? @rarkins

**Reply by rarkins** [reply]
I don't understand your example and question, so I'm not sure

**Comment by viceice** [comment]
renovate will update all deps by default, so you need a package rule to disable all deps and exclude the one you want

**Comment by rafat-parveen** [comment]
We want to run renovate for dependecies scanning based on the Event.

The Event would be when any new tag is generated for particular repo then all the place where the repo is referenced, the tag shoud be updated there.

_**Example**_:
let say one repository name:  `terraform-aws-s3-bucket` I have updated some code on it and once my PR is merged, it generate new tag.
once any new tag is generated for this repository, I have to update it's new tag in all the repo where this **terraform-aws-s3-bucket** is used. ( suppose new tag now v2.0.0).

another repository called, `aws-static-site-blueprint`
has multiple terraform file, **s3-bucket.tf**, **lamda.tf**, **xyz.tf**

In **s3-bucket.tf**, we use module to reuse the code from other repos (**terraform-aws-s3-bucket** ) and called it like this,
`module “s3” {
source = "git@gitlab.com:repo/blueprint/terraform-aws-s3-bucket?ref=v1.0.0"
}`

**lamda.tf**
`module “lamda” {
source = "git@gitlab.com:repo/blueprint/terraform-aws-lamda?ref=v1.0.0"
}`

so now I want to update the tag on s3 to v2.0.0 as this s3 repo generated new tag by my PR merge, so I want to automate this update via renovate, but renovate update the tag in all `.tf` file where it find update.

so I want to filter based on the source of terrfaorm module where dependencies shoud be updated based on which repo is merged and generated new tag at the moment.

@viceice, @rarkins  Is there any way I can achieve this or If I need to use regex pattern to match then how it should be look like so I can generalize for any event trigger.

let me know if you need more clarity on the question.

**Reply by rarkins** [reply]
That is not possible/supported
